src/modules/user/user.dao.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { User } from '../../../entities/User.entity';
+
+@Injectable()
+export class UserDAO {
+  constructor(
+    @InjectRepository(User)
+    private readonly repository: Repository<User>
+  ) {}
+
+  async getUserInfoById(userId: string): Promise<User> {
+    return this.repository
+      .createQueryBuilder('user')
+      .leftJoinAndSelect('user.company', 'company')
+      .where('user.id = :userId', { userId })
+      .getOne();
+  }
+
+  async getUserByEmail(email: string): Promise<User> {
+    return this.repository
+      .createQueryBuilder('user')
+      .where('user.email = :email', { email })
+      .getOne();
+  }
```

Comment:
@matheo97 is there a function that creates the user? shouldn't we include it among this functions as well?

Reply:
We don't have it but we definetely need it, I propose you to approve this PR and I can work on the CRUD of users as my next task, this way you guys will have the 'session' already available when working in your endpoints.